### PR TITLE
feat(agw) add management server and sync 1588 parameters

### PR DESCRIPTION
**Pull request by sxkmg**
Signed-off-by: root <369888042@qq.com>

 feat(agw): add management server and sync 1588 parameters
   

## Summary

add None 243  management server and sync 1588 parameters

## Test Plan

1.Start the magma service
2.Configure the base station none-243 to connect to the gateway
3.Test and issue parameters through the magma swagger API(https://magma-test.localhost:9443/swagger/v1/ui/lte#/EnodeBs/get_lte__network_id__enodebs__enodeb_serial_)
image

4.View the configuration file and verify whether the parameters are distributed to the gateway
（ sudo tail -1000f /var/opt/magma/configs/gateway.mconfig ）
image

5.Check whether the parameters are successfully distributed to the base station
sudo tail -1000f /var/log/enodebd.log
image

image

### CODE REVIEW

File: orc8r/cloud/docker/.env [hunk 1]
```diff
@@ -10,3 +10,4 @@
 # limitations under the License.
 
 COMPOSE_PROJECT_NAME=orc8r
+COMPOSE_FILE=docker-compose.yml:docker-compose.metrics.yml:docker-compose.override.yml
```
Comment (uri200) [inline orc8r/cloud/docker/.env]: i think you need to remove this

File: orc8r/cloud/docker/run.py [hunk 2]
```diff
@@ -21,7 +21,8 @@
 from typing import List
 
 HOST_BUILD_CTX = '/tmp/magma_orc8r_build'
-DO_NOT_COMMIT = '# DO NOT COMMIT THIS CHANGE'
+#DO_NOT_COMMIT = '# DO NOT COMMIT THIS CHANGE'
+DO_NOT_COMMIT = ''
```
Comment (uri200) [inline orc8r/cloud/docker/run.py]: remove these please

**Review by uri200 — CHANGES_REQUESTED** [review]

File: orc8r/cloud/docker/.env (hunk 1 above)
Comment (sxkmg) [reply orc8r/cloud/docker/.env]: OK, it has been deleted, thank you!

File: orc8r/cloud/docker/run.py (hunk 2 above)
Comment (sxkmg) [reply orc8r/cloud/docker/run.py]: OK, it has been deleted, thank you!

**Comment by uri200** [comment]
you may need to review the check that faile. For example you will need to change the name of your PR and ad `:`
```
feat(agw):
```
to pass the semantic check

You may need to rebase to make the other test pass

@andreilee @hcgatewood  please take a look at this PR

**Review by hcgatewood — APPROVED** [review]
Deferring to @ulaskozat / @uri200 for domain-specific Orc8r functionality

File: lte/cloud/go/services/lte/obsidian/models/swagger.v1.yml [hunk 3]
```diff
@@ -2292,6 +2292,67 @@ definitions:
...
+        example: 300
+      sync_1588_unicast_enable:
+        type: boolean
+        example: true
+      sync_1588_unicast_serverIp:
+        type: string
+        format: ipv4
+        example: '0.0.0.0'
+
+  managementServer:
```
Comment (andreilee) [inline lte/cloud/go/services/lte/obsidian/models/swagger.v1.yml:2510]: Mind the naming conventions
```suggestion
  management_erver:
```

File: lte/cloud/go/services/lte/obsidian/models/swagger.v1.yml [hunk 4]
```diff
@@ -2292,6 +2292,67 @@ definitions:
...
+    description: ManagementServer Configuration for  eNodeb devices.
```
Comment (andreilee) [inline lte/cloud/go/services/lte/obsidian/models/swagger.v1.yml:2511]:
```suggestion
    description: Management server configuration for eNodeb devices.
```

File: lte/cloud/go/services/lte/obsidian/models/swagger.v1.yml [hunk 5]
```diff
@@ -2292,6 +2292,67 @@ definitions:
...
+      mme_pool_2:
+        type: string
+        format: ipv4
+        example: ''
+      managementServer:
+        $ref: '#/definitions/managementServer'
+      sync_1588:
+        $ref: '#/definitions/sync_1588'
+  sync_1588:
+    description: sync_1588 Configuration for  eNodeb devices.
```
Comment (andreilee) [inline lte/cloud/go/services/lte/obsidian/models/swagger.v1.yml:2476]:
```suggestion
    description: sync_1588 Configuration for eNodeb devices.
```

Comment (andreilee) [reply lte/cloud/go/services/lte/obsidian/models/swagger.v1.yml:2476]: Looks like there's a few versions - is there a reference that can be included to a specification number?

File: lte/cloud/go/services/lte/obsidian/models/swagger.v1.yml [hunk 6]
```diff
@@ -2292,6 +2292,67 @@ definitions:
...
+    type: object
+    properties:
+      sync_1588_switch:
+        type: boolean
+        example: false
+      sync_1588_domain:
+        type: integer
+        format: int32
+        example: 0
+      sync_1588_msg_interval:
```
Comment (andreilee) [inline lte/cloud/go/services/lte/obsidian/models/swagger.v1.yml:2486]: include the units, definition (as well as info to look up with a spec), and limits

File: lte/cloud/go/services/lte/obsidian/models/swagger.v1.yml [hunk 7]
```diff
@@ -2292,6 +2292,67 @@ definitions:
...
+        format: uint32
+        example: 300
+      sync_1588_asymmetry:
+        type: integer
+        format: uint32
+        example: 300
+      sync_1588_unicast_enable:
+        type: boolean
+        example: true
+      sync_1588_unicast_serverIp:
```
Comment (andreilee) [inline lte/cloud/go/services/lte/obsidian/models/swagger.v1.yml:2505]: Please mind the naming conventions
```suggestion
      sync_1588_unicast_server_ip:
```

File: lte/cloud/go/services/lte/obsidian/models/swagger.v1.yml (hunk 6 above)
Comment (andreilee) [reply lte/cloud/go/services/lte/obsidian/models/swagger.v1.yml:2486]: Please also extend this to the other properties in the swagger definitions.

File: lte/cloud/go/services/lte/obsidian/models/swagger.v1.yml (hunk 4 above)
Comment (andreilee) [reply lte/cloud/go/services/lte/obsidian/models/swagger.v1.yml:2511]: Also include information about this being for a TR-069 auto configuration server

File: nms/packages/magmalte/app/views/equipment/EnodebDetailConfigEdit.js [hunk 8]
```diff
@@ -665,6 +928,83 @@ function buildRanConfig(config: enodeb_configuration, optConfig: OptConfig) {
...
+  if (optConfig.sync_1588_unicast_serverIp !== '') {
+    response.sync_1588.sync_1588_unicast_serverIp = String(
+      optConfig.sync_1588_unicast_serverIp,
+    );
+  }
+  if (optConfig.sync_1588_unicast_enable !== '') {
+    response.sync_1588.sync_1588_unicast_enable = Boolean(
+      optConfig.sync_1588_unicast_enable,
+    );
+  }
```
Comment (andreilee) [inline nms/packages/magmalte/app/views/equipment/EnodebDetailConfigEdit.js]: There's a lot of repetitive code added here. Can this be done in a way to adhere to the "DRY" principle?

**Review by andreilee — CHANGES_REQUESTED** [review]

**Comment by andreilee** [comment]
I see that there are 72 commits included in this pull request. Please rebase your branch, and merge commits. Use git amend to update your commits.

**Review by talkhasib — CHANGES_REQUESTED** [review]
We should not be changing the management server as this might break TR69 management of the eNB using Magma AGW.
Magma AGW uses DNS to configure itself as the management server.

**Comment by sxkmg** [comment]
ee

**Pull request closed without merging**